src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Automation/Peers/MenuItemAutomationPeer.cs

```diff
@@ -104,7 +104,10 @@ override protected int GetSizeOfSetCore()
 
                 foreach (var item in parent.Items)
                 {
-                    if (item is Separator)
+                    // ItemAutomationPeer.GetSizeOfSetFromItemsControl excludes invisible UI elements
+                    // from its count, so we only exclude separators that are visible. Otherwise, we would
+                    // double count invisible separators, leading to a lower count than expected.
+                    if (item is Separator separator && separator.Visibility == Visibility.Visible)
```

Comment:
```suggestion
                    if (item is Separator { Visibility: Visibility.Visible })
```
and save a local declaration

Reply:
What would the local declaration be used for in this scenario?

Reply:
Or did you mean taking this suggestion saves the local declaration?

Reply:
Exactly